rust/src/jsonbuilder.rs

```diff
@@ -698,6 +698,15 @@ impl JsonBuilder {
         }
         Ok(())
     }
+
+    fn encode_base64(&mut self, val: &[u8]) -> Result<&mut Self, JsonError> {
+        // Before base64 encoding, make sure enough space is reserved.
+        if self.buf.capacity() < self.buf.len() + val.len() * 2 {
+            self.buf.try_reserve(val.len() * 2)?;
```

Comment:
base64 is rather ((val.len()+2)/3)*4 no ?

Reply:
like one byte gets encoded as 4 bytes

Reply:
Ah yeah, mines too small.  `base64` has a function for this, will just use that.